core_data_modules/traced_data/util/fold_traced_data.py

```diff
@@ -1,90 +1,77 @@
-        :param fold_fn: Function to use to fold each pair of TracedData objects.
-        :type fold_fn: function of (TracedData, TracedData) -> TracedData
-        :return: Folded TracedData objects.
-        :rtype: iterable of TracedData
+        Concatenates x and y, separating them by a semicolon.
+        
+        If one of the values is None, the other value is returning without adding a semicolon anywhere.
+        
+        :param x: Value to fold.
+        :type x: str | None
+        :param y: Value to fold.
+        :type y: str | None
+        :return: x and y concatenated together, with a semicolon separating them.
+        :rtype: str | None
         """
-        folded_data = []
-
-        for group in groups:
-            folded_td = group.pop(0)
-            while len(group) > 0:
-                folded_td = fold_fn(folded_td, group.pop(0))
-            folded_data.append(folded_td)
+        if x is None:
+            return y
+        if y is None:
+            return x
 
-        return folded_data
+        return f"{x};{y}"
 
-    @staticmethod
-    def assert_equal_keys_equal(td_1, td_2, equal_keys):
+    @classmethod
+    def _is_control_code(cls, code):
         """
-        Checks that the provided TracedData objects contain exactly the same values for each of the provided keys.
-
-        Raises an AssertionError if mis-matching keys are found, otherwise returns no value and has no side-effect.
-
-        :param td_1: TracedData object to check for value equality.
-        :type td_1: TracedData
-        :param td_2: TracedData object to check for value equality.
-        :type td_2: TracedData
-        :param equal_keys: Keys to check for equality in the TracedData objects.
-        :type equal_keys: iterable of str
+        Check if the given code string is a control code.
+        
+        :param code: Code string to check.
+        :type code: str
+        :return: True if the code is a control code, False otherwise.
+        :rtype: bool
         """
-        for key in equal_keys:
-            assert td_1.get(key) == td_2.get(key), "Key '{}' should be the same in both td_1 and td_2 but is " \
-                                                   "different (has values '{}' and '{}' " \
-                                                   "respectively)".format(key, td_1.get(key), td_2.get(key))
+        # TODO: If we move this function to `Codes`, that will reduce the risk of us updating Core but not updating
+        #       the body of this function.
+        return code in cls.CONTROL_CODES
 
     @staticmethod
-    def _is_control_code(code):
-        return code in {
-            Codes.STOP, Codes.CODING_ERROR, Codes.NOT_REVIEWED, Codes.NOT_INTERNALLY_CONSISTENT,
-            Codes.NOT_CODED, Codes.TRUE_MISSING, Codes.SKIPPED, Codes.WRONG_SCHEME, Codes.NOISE_OTHER_CHANNEL, None
-        }
-
-    @staticmethod
-    def reconcile_missing_values(value_1, value_2):
+    def control_code(x, y):
```

Comment:
consider `control_code_by_precedence`

`FoldingStrategies.ControlCode` doesn't really say much to me about what the method is going to do

Reply:
Yeah, I found it quite hard coming up "action" names for all the methods here. control_code_by_precedence is a good suggestion, I've changed that one